src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs

```diff
@@ -480,35 +174,67 @@ public async Task InitializeFromProfile(DacDeployOptions options, string profile
             SetOptions(options);
         }
 
+        /// <summary>
+        /// Populates OptionsMapTable with the boolean type properties
+        /// </summary>
+        /// <param name="options"></param>
+        public void PopulateOptionsMapTable(DacDeployOptions options)
+        {
+            // To fill the options map table directly fomr the boolean type DacDeployoptions
+            PropertyInfo[] dacDeploymentOptionsProperties = options.GetType().GetProperties();
+            foreach (var prop in dacDeploymentOptionsProperties)
+            {
+                if (prop.PropertyType == typeof(System.Boolean))
+                {
+                    object setProp = GetDeploymentOptionProp(prop, options);
+                    var displayNameAttribute = prop.GetCustomAttributes<DisplayNameAttribute>().FirstOrDefault();
+                    this.OptionsMapTable[displayNameAttribute.DisplayName] = (DeploymentOptionProperty<bool>)setProp;
+                }
+            }
+        }
+
         public void SetOptions(DacDeployOptions options)
         {
             System.Reflection.PropertyInfo[] deploymentOptionsProperties = this.GetType().GetProperties();
+            // Set the default options properties
+            PopulateOptionsMapTable(options);
+            SetGenericDacDeploymentProperties(options);
+        }
 
+        /// <summary>
+        /// Preparing all non boolean properties (except optionsMapTable)
+        /// </summary>
+        /// <param name="options"></param>
+        public void SetGenericDacDeploymentProperties(DacDeployOptions options)
+        {
+            // preparing remaining properties
+            PropertyInfo[] deploymentOptionsProperties = this.GetType().GetProperties();
             foreach (var deployOptionsProp in deploymentOptionsProperties)
             {
-                var prop = options.GetType().GetProperty(deployOptionsProp.Name);
-                // Note that we set excluded object types here since dacfx has this value as null;
-                if (prop != null)
+                if (deployOptionsProp.Name != "OptionsMapTable")
                 {
-                    SetGenericDeployOptionProps(prop, options, deployOptionsProp);
+                    var prop = options.GetType().GetProperty(deployOptionsProp.Name);
+                    object setProp = GetDeploymentOptionProp(prop, options);
+                    deployOptionsProp.SetValue(this, setProp);
                 }
             }
         }
 
         /// <summary>
-        /// Sets the Value and Description to all properties
+        /// Prepares and returns the Value and Description to all properties
         /// </summary>
         /// <param name="prop"></param>
         /// <param name="val"></param>
         /// <param name="deployOptionsProp"></param>
```

Comment:
these doc comments are out of date and don't match the actual parameters. Also, please add descriptions for these params